.github/workflows/ruleset.yml

```diff
@@ -0,0 +1,90 @@
+name: Ruleset checks
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  # Makes sure the rulesets don't throw unexpected errors or warnings.
+  # This workflow needs to be run against a high PHP version to prevent triggering the syntax error check.
+  # It also needs to be run against all PHPCS versions WPCS is tested against.
+  #
+  # Performs the following steps:
+  # - Checks out the repository.
+  # - Configures caching for Composer.
+  # - Sets up PHP.
+  # - Installs PHPCS.
+  # - Installs Composer dependencies.
+  # - Checks the WordPress rulesets.
+  ruleset-checks:
+    name: PHP ${{ matrix.php }} on PHPCS ${{ matrix.phpcs_branch }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php: [ '7.4' ]
```

Comment:
This will always be just 7.4, or was this added here for a future use case if we want to add different PHP versions?

Reply:
I don't think there was any particular reason. We could remove it in favor of hard coding the version below, or leave it in case we add other versions for testing. It probably makes it more clear at the very beginning of the job which versions are used for testing. But no strong preference either way.